core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -38,8 +39,8 @@ class MapUnit {
     var mapUnitAction : MapUnitAction? = null
     var action: String? // work, automation, fortifying, I dunno what.
         // getter and setter for compatibility: make sure string-based actions still work
-        get() = mapUnitAction?.name
-        set(value) { mapUnitAction = value?.let{ MapUnitAction(this, value) } }
+        get() = (mapUnitAction as? StringAction)?.action ?: mapUnitAction?.let { "" } // null if no action is assigned.
```

Comment:
Man, this is some serious voodoo.
This can be made much, much more readable. Without .let, without elvis operators. Just ifs.
A big part of making open source is that the future contributors will be able to make changes, and who's going to want to touch this?

Reply:
You're right 😅 it would leave some more space for comments well.

Could you replace with this? I'm on my phone only now 🙂

```Kotlin
get() {
  // StringAction wraps string-encoded commands
  if(mapUnitAction is StringAction)
    return mapUnitAction.action
  // any other unit action does count as a unit action, thus is not null. The actual logic is not string encoded, but by extending MapUnitAction
  if(mapUnitAction!=null)
    return ""
  // unit has no action
  return null
}
```

Also, would you like to make MapUnitAction abstract?